venues/NeurIPS.cc/2023/submission_preprocess.js

```diff
@@ -0,0 +1,71 @@
+async function process(client, edit, invitation) {
+    client.throwErrors = true
+    
+    const { note } = edit
+
+    if (note.ddate) { 
+      return
+    }
+
+    const profiles = await client.tools.getProfiles(note.content.authorids.value)
+    const profileIds = profiles.map(profile => profile.id)
+
+    const primaryAuthorField = note.content.corresponding_author.value
+    const primaryAuthorProfile = await client.tools.getProfiles([primaryAuthorField])
+    const primaryAuthor = primaryAuthorProfile ? primaryAuthorProfile[0].id : primaryAuthorField
+    if (!profileIds.includes(primaryAuthor)) {
+      return Promise.reject(new OpenReviewError({ name: 'Error', message: 'Select a paper co-author as corresponding author, ' + primaryAuthorField + ' does not appear in the author list' }))
+    }
+
+    if (note.content.financial_support) {
+      const financialSupportField = note.content.financial_support.value
```

Comment:
we need to be careful here because the `financial_support` can exist but the value is empty. 

This happens when they put a value and then they edit it clearing the field, then value is an empty string.

Reply:
I tried doing this in my local and there is no issue if the value is cleared

Reply:
Really? That's odd, I though it was going to fail here :
```js
const financialSupport = financialSupportProfile[0].id
```

Reply:
it does not go into the if, so we never make it to that code. Maybe I am doing something wrong and not understading the issue here

Reply:
what if `note.content.financial_support.value === ''`? Is this even possible?